Firestore/Source/API/FIRFirestore.mm

```diff
@@ -175,7 +180,7 @@ + (instancetype)firestoreForApp:(FIRApp *)app database:(NSString *)database {
 - (instancetype)initWithProjectID:(NSString *)projectID
                          database:(NSString *)database
                    persistenceKey:(NSString *)persistenceKey
-              credentialsProvider:(id<FSTCredentialsProvider>)credentialsProvider
+              credentialsProvider:(CredentialsProvider *)credentialsProvider
```

Comment:
(see other comments about unique_ptr. There's other instances of this too through this PR; I'll refrain from commenting on the rest of them.)